src/api.ts

```diff
@@ -301,13 +293,15 @@ async function getRemoteBranches(): Promise< Map< string, string > > {
 		);
 
 		repo.free();
-		// gc the repo
-		try {
-			await promisify( exec )( 'git gc', {
-				cwd: calypsoDir,
-			} );
-		} catch ( err ) {
-			l.error( { err }, 'git gc failed' );
+		// gc the repo if no builds are running
+		if ( pendingHashes.size === 0 ) {
+			try {
+				await promisify( exec )( 'git gc', {
```

Comment:
The fact that we only log on error makes it tricky to know whether we're doing this without error or if we never do it.

Reply:
always logging just clogs up the logs. we do this pretty often. maybe too often.